File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

double count_letters(string text);
double count_words(string text);
double count_sentences(string text);
void round_grade (int grade_level);

int main(void)
{
    string text = get_string("Text: ");
    double letters = count_letters(text);
    double words = count_words(text);
    double sentences = count_sentences(text);

    double L =  letters / words * 100;
    double S = sentences / words * 100;
    double index = 0.0588 * L - 0.296 * S - 15.8;

    int grade_level = round(index);

    round_grade(grade_level);
}


double count_letters(string text)
{
    int letters = 0;

    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalpha(text[i]))
        {
            letters += 1;
        }
    }
    return letters;
}


double count_words(string text)
{
    int words = 1;

    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isspace(text[i]))
        {
            words += 1;
        }
    }
    return words;
}


double count_sentences(string text)
{
    int sentences = 0;

    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == '.')
        {
            sentences += 1;
        }
        if (text[i] == '!')
        {
            sentences += 1;
        }
        if (text[i] == '?')
        {
            sentences += 1;
        }
    }
    return sentences;
}


void round_grade (int grade_level)
{
    if (grade_level < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade_level > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade_level);
    }
}

File: sentimental-readability/readability.py
def main():
    text = input("Text: ")

    letters = count_letters(text)
    words = count_words(text)
    sentences = count_sentences(text)

    L = letters / words * 100
    S = sentences / words * 100
    index = 0.0588 * L - 0.296 * S - 15.8

    grade_level = round(index)

    updated_level(grade_level)

def count_letters(text):
    letters_counter = 0

    for i in range(len(text)):
        if (text[i].isalpha() == True):
            letters_counter += 1
    return letters_counter

def count_words(text):
    words_counter = 1

    for i in range(len(text)):
        if (text[i].isspace() == True):
            words_counter += 1
    return words_counter

def count_sentences(text):
    sentence_counter = 0

    punctuation = ['.', '!', '?']

    for i in range(len(text)):
        if (text[i] in punctuation):
            sentence_counter += 1
    return sentence_counter

def updated_level(grade_level):

    if (grade_level < 1):
        print("Before Grade 1")

    elif (grade_level > 16):
        print("Grade 16+")

    else:
        print(f"Grade: {grade_level}")



main()